File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.HashMap;
import java.util.Map;

class ResistorColorDuo {

    private static Map<String, Integer> colorValues = new HashMap<>();
    static {
        colorValues.put("black", 0);
        colorValues.put("brown", 1);
        colorValues.put("red", 2);
        colorValues.put("orange", 3);
        colorValues.put("yellow", 4);
        colorValues.put("green", 5);
        colorValues.put("blue", 6);
        colorValues.put("violet", 7);
        colorValues.put("grey", 8);
        colorValues.put("white", 9);
    }

    int value(String[] colors) {
        return colorValues.get(colors[0]) * 10 + colorValues.get(colors[1]);
    }
}

File: ruby/resistor-color-duo/resistor_color_duo.rb
# frozen_string_literal: true

BAND_VALUES = { 'black' => 0, 'brown' => 1, 'red' => 2, 'orange' => 3, 'yellow' => 4,
                'green' => 5, 'blue' => 6, 'violet' => 7, 'grey' => 8, 'white' => 9 }.freeze

class ResistorColorDuo
  # Assumes at least two colors in bands
  def self.value(bands)
    BAND_VALUES[bands[0]] * 10 + BAND_VALUES[bands[1]]
  end
end
